app/src/test/java/org/simple/clinic/settings/changelanguage/ChangeLanguageUpdateTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.simple.clinic.settings.changelanguage
+
+import com.spotify.mobius.test.NextMatchers.hasEffects
+import com.spotify.mobius.test.NextMatchers.hasModel
+import com.spotify.mobius.test.NextMatchers.hasNoEffects
+import com.spotify.mobius.test.NextMatchers.hasNoModel
+import com.spotify.mobius.test.UpdateSpec
+import com.spotify.mobius.test.UpdateSpec.assertThatNext
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.settings.Language
+
+@RunWith(JUnitParamsRunner::class)
+class ChangeLanguageUpdateTest {
+
+  private val defaultModel = ChangeLanguageModel.FETCHING_LANGUAGES
+
+  private val englishIndia = Language(displayName = "English", languageCode = "en_IN")
+  private val hindiIndia = Language(displayName = "हिंदी", languageCode = "hi_IN")
+
+  private val spec = UpdateSpec<ChangeLanguageModel, ChangeLanguageEvent, ChangeLanguageEffect>(ChangeLanguageLogic::update)
+
+  @Test
+  @Parameters(method = "params for initial models when loading selected language")
+  fun `when the selected language is loaded, the model must be updated`(model: ChangeLanguageModel) {
+    spec
+        .given(model)
+        .whenEvent(CurrentSelectedLanguageLoadedEvent(englishIndia))
+        .then(assertThatNext(
+            hasModel(model.withCurrentLanguage(englishIndia)),
+            hasNoEffects()
+        ))
+  }
+
+  @Suppress("Unused")
+  private fun `params for initial models when loading selected language`(): List<ChangeLanguageModel> {
+    return listOf(
+        defaultModel,
+        defaultModel.withSupportedLanguages(listOf(englishIndia, hindiIndia))
+    )
+  }
+
+  @Test
+  @Parameters(method = "params for initial models when loading supported languages")
+  fun `when the supported languages are loaded, the model must be updated`(model: ChangeLanguageModel) {
```

Comment:
Does this have to be a parameterized test?

Reply:
See earlier comment; I'll break this into multiple tests.

Reply:
I tried breaking it up, but realised the alternate test isn't providing much value. Decided not to do it.